### PR TITLE
fix: refactor cmake project to deal with robot model dependencies

### PR DESCRIPTION
**Pull request by domire8**
## Description

This PR solves an issue where downstream projects had problems using the robot model library. 
Thanks @LouisBrunner for the effort to clean this up!

## Review guidelines

Estimated Time of Review: 5 minutes

## Checklist before merging:

- [x] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

File: source/control_librariesConfig.cmake.in [hunk 1]
```diff
@@ -1,34 +1,58 @@
...
     set_and_check(control_libraries_FOUND False)
     set_and_check(control_libraries_NOT_FOUND_MESSAGE "Unsupported component: ${_comp}")
   endif()
 
-  # Find robot model dependencies if the corresponding components are explicitly listed
-  if (${_comp} STREQUAL "controllers" OR ${_comp} STREQUAL "robot_model")
+  # Find robot model dependencies if the corresponding component is needed
+  # FIXME: this should be done automatically
+  if (${_comp} STREQUAL "robot_model")
     find_dependency(pinocchio)
```
Comment (domire8) [inline source/control_librariesConfig.cmake.in:28]: Do we need hpp-fcl here?

Comment (LouisBrunner) [reply source/control_librariesConfig.cmake.in:28]: Not sure. It built and ran correctly for me so I imagine `pinocchio` might be pulling it itself?

**Comment by domire8** [comment]
I've tested in in a backend and it seems to do the job!

**Review by yrh012 — APPROVED** [review]
Tested and looks great! Thank you

**Review by eeberhard — APPROVED** [review]
Thanks for all the efforts! I trust the issues have been resolved based on all of your testing.

**Pull request merged**